### ISSUE TITLE
Pin dependencies and add integrity checks

### ISSUE THREAD

**Issue by ewan-escience** [NLeSC/guide#344]
Linked dependencies, as in `<script src="//cdn.jsdelivr.net/npm/docsify/lib/plugins/search.min.js"></script>` don't have a version in their URL, meaning they will always point to the latest version. (Side note: why do these dependencies start with `//` instead of `https://`?). This has two risks:

1. A new version might break the guide (through a bug or a changed API)
2. A version with malicious code might be used

If you take off the last part of the path of these dependcies (e.g. https://cdn.jsdelivr.net/npm/docsify/lib/plugins/), you can navigate through the versions.

Furthermore, we should use [integrity checks](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) to ensure that the loaded code has not been altered with.

**Comment by bouweandela** [comment]
If we pin those dependencies, is there some way to automate updating them so we don't end up with very outdated dependencies one or two years from now?

**Comment by ewan-escience** [comment]
The point is to *not* update them automatically. But this would indeed require some regular maintenance.

If you don't think it's worth the effort, that's also fine, but I wanted to point out the risks we currently have.

**Comment by bouweandela** [comment]
I was thinking of automatically creating a pull request to update the version, so we can review if anything is broken. Would something like that be possible without too much effort?

**Comment by ewan-escience** [comment]
From [this issue](https://github.com/dependabot/dependabot-core/issues/3228), this does not seem to be supported (yet) by dependabot and I haven't found any other tool that can do this, so it looks like we would have to write a tool ourselves if we want automatic pull requests.

**Comment by bouweandela** [comment]
From https://github.com/dependabot/dependabot-core/issues/3228#issuecomment-1090409479 it looks like [renovte](https://docs.renovatebot.com/) supports it? Or is that a mistake?

**Comment by ewan-escience** [comment]
Hmm, if I understand correctly from [this](https://docs.renovatebot.com/modules/manager/cdnurl/) and [this](https://docs.renovatebot.com/modules/datasource/cdnjs/), [cdnjs](https://cdnjs.com/) is supported, but we use [jsDelivr](https://www.jsdelivr.com/), which is [apparently not yet supported](https://github.com/renovatebot/renovate/issues/26936). Maybe we can switch to cdnjs. If we did though, would you be in favour of adding Renovate as a dependency checker to this project?

**Comment by bouweandela** [comment]
Yes, I think so. It looks like https://docs.renovatebot.com/modules/manager/html/ may be preferred over https://docs.renovatebot.com/modules/manager/cdnurl/ if you would like to add integrity checks:

> This [cdnurl] manager does not read or update Subresource Integrity (SRI) hashes. It replaces any matching URL it finds, without adding or updating script integrity hashes. We recommend you use the `html` manager if you need SRI updating.
